### PR TITLE
Generate synthetics documentation

### PR DESCRIPTION
**Pull request by nmuesch**
Use tfplugindocs (see #798) to generate the synthetics documentation

### CODE REVIEW

File: examples/resources/datadog_synthetics_test/resource.tf [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Example Usage (Synthetics API test)
+# Create a new Datadog Synthetics API/HTTP test on https://www.example.org
+resource "datadog_synthetics_test" "test_api" {
+  type = "api"
+  subtype = "http"
+  request = {
+    method = "GET"
+    url = "https://www.example.org"
+  }
```
Comment (zippolyte) [inline examples/resources/datadog_synthetics_test/resource.tf:9]:
```suggestion
  request_definition {
    method = "GET"
    url = "https://www.example.org"
  }
```

File: examples/resources/datadog_synthetics_test/resource.tf [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  status = "live"
+}
+
+
+# Example Usage (Synthetics SSL test)
+# Create a new Datadog Synthetics API/SSL test on example.org
+resource "datadog_synthetics_test" "test_ssl" {
+  type = "api"
+  subtype = "ssl"
+  request = {
```
Comment (zippolyte) [inline examples/resources/datadog_synthetics_test/resource.tf:45]:
```suggestion
  request_definition {
```

File: examples/resources/datadog_synthetics_test/resource.tf [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  status = "live"
+}
+
+
+# Example Usage (Synthetics TCP test)
+# Create a new Datadog Synthetics API/TCP test on example.org
+resource "datadog_synthetics_test" "test_tcp" {
+  type = "api"
+  subtype = "tcp"
+  request = {
```
Comment (zippolyte) [inline examples/resources/datadog_synthetics_test/resource.tf:72]:
```suggestion
  request_definition {
```

File: examples/resources/datadog_synthetics_test/resource.tf [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  status = "live"
+}
+
+
+# Example Usage (Synthetics DNS test)
+# Create a new Datadog Synthetics API/DNS test on example.org
+resource "datadog_synthetics_test" "test_dns" {
+  type = "api"
+  subtype = "dns"
+  request = {
```
Comment (zippolyte) [inline examples/resources/datadog_synthetics_test/resource.tf:98]:
```suggestion
  request_definition {
```

File: examples/resources/datadog_synthetics_test/resource.tf [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+
+# Example Usage (Synthetics Browser test)
+# Support for Synthetics Browser test steps is limited (see below)
+# Create a new Datadog Synthetics Browser test starting on https://www.example.org
+resource "datadog_synthetics_test" "test_browser" {
+  type = "browser"
+
+  request = {
```
Comment (zippolyte) [inline examples/resources/datadog_synthetics_test/resource.tf:125]:
```suggestion
  request_definition {
```

**Review by zippolyte — CHANGES_REQUESTED** [review]

File: examples/resources/datadog_synthetics_test/resource.tf (hunk 1 above)
Comment (nmuesch) [reply examples/resources/datadog_synthetics_test/resource.tf]: Thanks, sorry I keep missing these 🤦

**Review by zippolyte — APPROVED** [review]

**Comment by nmuesch** [comment]
Thanks, no go files were modified so going to merge.

**Pull request merged**